### ISSUE TITLE
trouble getting sample app to work

### ISSUE THREAD

**Issue by vesper8** [rigor789/nativescript-vue#27]
am trying to get the sample app to work and running into these issues:

am on mac, latest sierra, running directly from the mac's OS
npm 5.0.4
node 8.1.3
nativescript 3.1

```
git clone https://github.com/rigor789/nativescript-vue.git
cd nativescript-vue

~/nativescript-vue/npm i
~/nativescript-vue/npm link

cd samples
~/nativescript-vue/samples/npm link nativescript-vue
~/nativescript-vue/samples/tns run android --syncAllFiles
```

I do this with my genymotion opened and my Google Pixel 7.1.0 emulator launched and ready

It goes through a long build process, downloading gradle etc.. but then it always fails with these messages (see at bottom):

```
Support for Node.js 8.1.3 is not verified. This CLI might not install or run properly.

Searching for devices...

> nativescript-dev-android-snapshot@0.0.9 postinstall /nativescript-vue/samples/node_modules/nativescript-dev-android-snapshot
> node postinstall.js

Hook already installed: nativescript-dev-android-snapshot
Hook already installed: nativescript-dev-android-snapshot
npm
 WARN optional SKIPPING OPTIONAL DEPENDENCY: wrappy@1.0.2 (node_modules/wrappy):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/wrappy' -> '/nativescript-vue/samples/node_modules/wrappy'

npm
 WARN
optional
 SKIPPING OPTIONAL DEPENDENCY: to-fast-properties@1.0.3 (node_modules/to-fast-properties):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/to-fast-properties' -> '/nativescript-vue/samples/node_modules/to-fast-properties'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: supports-color@2.0.0 (node_modules/supports-color):
npm
WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/supports-color' -> '/nativescript-vue/samples/node_modules/supports-color'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: strip-ansi@3.0.1 (node_modules/strip-ansi):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/strip-ansi' -> '/nativescript-vue/samples/node_modules/strip-ansi'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: semver@5.3.0 (node_modules/semver):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/semver' -> '/nativescript-vue/samples/node_modules/semver'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: regenerator-runtime@0.10.5 (node_modules/regenerator-runtime):
npm WARN enoent
 SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/regenerator-runtime' -> '/nativescript-vue/samples/node_modules/regenerator-runtime'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: path-is-absolute@1.0.1 (node_modules/path-is-absolute):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/path-is-absolute' -> '/nativescript-vue/samples/node_modules/path-is-absolute'
npm
 WARN optional SKIPPING OPTIONAL DEPENDENCY: once@1.4.0 (node_modules/once):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/once' -> '/nativescript-vue/samples/node_modules/once'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: ms@2.0.0 (node_modules/ms):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/ms' -> '/nativescript-vue/samples/node_modules/ms'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: mkdirp@0.5.1 (node_modules/mkdirp):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/mkdirp' -> '/nativescript-vue/samples/node_modules/mkdirp'

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: minimist@0.0.8 (node_modules/minimist):
npm
WARN

enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/minimist' -> '/nativescript-vue/samples/node_modules/minimist'

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: minimatch@3.0.4 (node_modules/minimatch):

npm WARN
 enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/minimatch' -> '/nativescript-vue/samples/node_modules/minimatch'

npm
 WARN optional SKIPPING OPTIONAL DEPENDENCY: loose-envify@1.3.1 (node_modules/loose-envify):

npm WARN enoent
 SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/loose-envify' -> '/nativescript-vue/samples/node_modules/loose-envify'

npm WARN
optional SKIPPING OPTIONAL DEPENDENCY: lodash@4.17.4 (node_modules/lodash):
npm
WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/lodash' -> '/nativescript-vue/samples/node_modules/lodash'
npm WARN
optional
 SKIPPING OPTIONAL DEPENDENCY: invariant@2.2.2 (node_modules/invariant):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/invariant' -> '/nativescript-vue/samples/node_modules/invariant'

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: inherits@2.0.3 (node_modules/inherits):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/inherits' -> '/nativescript-vue/samples/node_modules/inherits'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: inflight@1.0.6 (node_modules/inflight):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/inflight' -> '/nativescript-vue/samples/node_modules/inflight'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: has-ansi@2.0.0 (node_modules/has-ansi):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/has-ansi' -> '/nativescript-vue/samples/node_modules/has-ansi'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: globals@9.18.0 (node_modules/globals):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/globals' -> '/nativescript-vue/samples/node_modules/globals'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esutils@2.0.2 (node_modules/esutils):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/esutils' -> '/nativescript-vue/samples/node_modules/esutils'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: escape-string-regexp@1.0.5 (node_modules/escape-string-regexp):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/escape-string-regexp' -> '/nativescript-vue/samples/node_modules/escape-string-regexp'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: debug@2.6.8 (node_modules/debug):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/debug' -> '/nativescript-vue/samples/node_modules/debug'
npm WARN

optional SKIPPING OPTIONAL DEPENDENCY: core-js@2.4.1 (node_modules/core-js):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/core-js' -> '/nativescript-vue/samples/node_modules/core-js'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: concat-map@0.0.1 (node_modules/concat-map):
npm WARN enoent
 SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/concat-map' -> '/nativescript-vue/samples/node_modules/concat-map'

npm
 WARN optional SKIPPING OPTIONAL DEPENDENCY: chalk@1.1.3 (node_modules/chalk):

npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/chalk' -> '/nativescript-vue/samples/node_modules/chalk'

npm WARN
optional SKIPPING OPTIONAL DEPENDENCY: brace-expansion@1.1.8 (node_modules/brace-expansion):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/brace-expansion' -> '/nativescript-vue/samples/node_modules/brace-expansion'

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: balanced-match@1.0.0 (node_modules/balanced-match):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/balanced-match' -> '/nativescript-vue/samples/node_modules/balanced-match'
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: babel-runtime@6.23.0 (node_modules/babel-runtime):
npm WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/babel-runtime' -> '/nativescript-vue/samples/node_modules/babel-runtime'

npm

WARN optional SKIPPING OPTIONAL DEPENDENCY: babel-messages@6.23.0 (node_modules/babel-messages):
npm WARN
 enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/babel-messages' -> '/nativescript-vue/samples/node_modules/babel-messages'

npm
 WARN optional SKIPPING OPTIONAL DEPENDENCY: babel-code-frame@6.22.0 (node_modules/babel-code-frame):

npm WARN
 enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/babel-code-frame' -> '/nativescript-vue/samples/node_modules/babel-code-frame'

npm
 WARN
 optional
 SKIPPING OPTIONAL DEPENDENCY: ansi-styles@2.2.1 (node_modules/ansi-styles):
npm
WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/ansi-styles' -> '/nativescript-vue/samples/node_modules/ansi-styles'

npm

WARN
optional
 SKIPPING OPTIONAL DEPENDENCY: ansi-regex@2.1.1 (node_modules/ansi-regex):
npm
WARN enoent SKIPPING OPTIONAL DEPENDENCY: ENOENT: no such file or directory, rename '/nativescript-vue/samples/node_modules/nativescript-vue/node_modules/ansi-regex' -> '/nativescript-vue/samples/node_modules/ansi-regex'



added 10 packages, removed 378 packages and moved 32 packages in 9.837s
added 32 packages and removed 1 package in 1.708s
Copying template files...
Installing  tns-android
  ◝ Installing tns-android+ tns-android@3.0.0
added 1 package in 1.164s
Project successfully created.
Executing before-prepare hook from /nativescript-vue/samples/hooks/before-prepare/nativescript-dev-android-snapshot.js
Preparing project...
Project successfully prepared (Android)
Executing after-prepare hook from /nativescript-vue/samples/hooks/after-prepare/nativescript-dev-android-snapshot.js
Building project...

:config phase:  createDefaultIncludeFiles

:config phase:  addAarDependencies
The JavaCompile.setDependencyCacheDir() method has been deprecated and is scheduled to be removed in Gradle 4.0.
The TaskInputs.source(Object) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use TaskInputs.file(Object).skipWhenEmpty() instead.
Incremental java compilation is an incubating feature.
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
        at build_6j22f0t661b1fo66alxcy5wbl.run(/nativescript-vue/samples/platforms/android/build-tools/android-static-binding-generator/build.gradle:126)
:preBuild UP-TO-DATE
:preDebugBuild UP-TO-DATE
:checkDebugManifest
:preReleaseBuild UP-TO-DATE
:prepareComAndroidSupportAnimatedVectorDrawable2510Library
:prepareComAndroidSupportAppcompatV72510Library
:prepareComAndroidSupportDesign2510Library
:prepareComAndroidSupportRecyclerviewV72510Library
:prepareComAndroidSupportSupportCompat2510Library
:prepareComAndroidSupportSupportCoreUi2510Library
:prepareComAndroidSupportSupportCoreUtils2510Library
:prepareComAndroidSupportSupportFragment2510Library
:prepareComAndroidSupportSupportMediaCompat2510Library
:prepareComAndroidSupportSupportV42510Library
:prepareComAndroidSupportSupportVectorDrawable2510Library
:prepareComAndroidSupportTransition2510Library
:prepareVuesampleRuntimeUnspecifiedLibrary
:prepareDebugDependencies
:compileDebugAidl
:compileDebugRenderscript
:generateDebugBuildConfig
:cleanLocalAarFiles UP-TO-DATE
:ensureMetadataOutDir UP-TO-DATE
:collectAllJars
:setProperties
:generateTypescriptDefinitions SKIPPED
:copyTypings SKIPPED
:asbg:generateInterfaceNamesList
:asbg:traverseJsFiles
:asbg:runAstParser
Running full build
gave the following error:
[ERROR] module.js:487
    throw err;
    ^

Error: Cannot find module 'js-tokens'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/nativescript-vue/samples/node_modules/babel-code-frame/lib/index.js:58:17)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)

:asbg:generateBindings
Exception in thread "main" java.io.FileNotFoundException: /nativescript-vue/samples/platforms/android/build-tools/android-static-binding-generator/bindings.txt (No such file or directory)
        at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(FileInputStream.java:195)
        at java.io.FileInputStream.<init>(FileInputStream.java:138)
        at java.io.FileInputStream.<init>(FileInputStream.java:93)
        at org.nativescript.staticbindinggenerator.Generator.getRows(Generator.java:123)
        at org.nativescript.staticbindinggenerator.Generator.generateBindings(Generator.java:66)
        at org.nativescript.staticbindinggenerator.Generator.writeBindings(Generator.java:50)
        at org.nativescript.staticbindinggenerator.Main.main(Main.java:15)
:asbg:generateBindings FAILED

FAILURE: Build failed with an exception.

* Where:
Build file '/nativescript-vue/samples/platforms/android/build-tools/android-static-binding-generator/build.gradle' line: 251

* What went wrong:
Execution failed for task ':asbg:generateBindings'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 5.08 secs
Unable to apply changes on device: 192.168.58.101:5555. Error is: Command ./gradlew failed with exit code 1.
```

Help would be much appreciated!

**Comment by vesper8** [comment]
I hadn't finished setting up nativescript and hadn't installed the tools. I did that now but still getting an error

```
tns run android

Support for Node.js 8.1.3 is not verified. This CLI might not install or run properly.

Searching for devices...
up to date in 0.836s
up to date in 0.458s
Executing before-prepare hook from /nativescript-vue/samples/hooks/before-prepare/nativescript-dev-android-snapshot.js
Preparing project...
Project successfully prepared (Android)
Executing after-prepare hook from /nativescript-vue/samples/hooks/after-prepare/nativescript-dev-android-snapshot.js
Building project...

:config phase:  createDefaultIncludeFiles

:config phase:  addAarDependencies
The JavaCompile.setDependencyCacheDir() method has been deprecated and is scheduled to be removed in Gradle 4.0.
The TaskInputs.source(Object) method has been deprecated and is scheduled to be removed in Gradle 4.0. Please use TaskInputs.file(Object).skipWhenEmpty() instead.
Incremental java compilation is an incubating feature.
The Task.leftShift(Closure) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use Task.doLast(Action) instead.
        at build_6j22f0t661b1fo66alxcy5wbl.run(/nativescript-vue/samples/platforms/android/build-tools/android-static-binding-generator/build.gradle:126)
:preBuild UP-TO-DATE
:preDebugBuild UP-TO-DATE
:checkDebugManifest
:preReleaseBuild UP-TO-DATE
:prepareComAndroidSupportAnimatedVectorDrawable2510Library UP-TO-DATE
:prepareComAndroidSupportAppcompatV72510Library UP-TO-DATE
:prepareComAndroidSupportDesign2510Library UP-TO-DATE
:prepareComAndroidSupportRecyclerviewV72510Library UP-TO-DATE
:prepareComAndroidSupportSupportCompat2510Library UP-TO-DATE
:prepareComAndroidSupportSupportCoreUi2510Library UP-TO-DATE
:prepareComAndroidSupportSupportCoreUtils2510Library UP-TO-DATE
:prepareComAndroidSupportSupportFragment2510Library UP-TO-DATE
:prepareComAndroidSupportSupportMediaCompat2510Library UP-TO-DATE
:prepareComAndroidSupportSupportV42510Library UP-TO-DATE
:prepareComAndroidSupportSupportVectorDrawable2510Library UP-TO-DATE
:prepareComAndroidSupportTransition2510Library UP-TO-DATE
:prepareVuesampleRuntimeUnspecifiedLibrary UP-TO-DATE
:prepareDebugDependencies
:compileDebugAidl UP-TO-DATE
:compileDebugRenderscript UP-TO-DATE
:generateDebugBuildConfig UP-TO-DATE
:cleanLocalAarFiles UP-TO-DATE
:ensureMetadataOutDir UP-TO-DATE
:collectAllJars
:setProperties
:generateTypescriptDefinitions SKIPPED
:copyTypings SKIPPED
:asbg:generateInterfaceNamesList
:asbg:traverseJsFiles
:asbg:runAstParser UP-TO-DATE
:asbg:generateBindings
Exception in thread "main" java.io.FileNotFoundException: /nativescript-vue/samples/platforms/android/build-tools/android-static-binding-generator/bindings.txt (No such file or directory)
        at java.io.FileInputStream.open0(Native Method)
        at java.io.FileInputStream.open(FileInputStream.java:195)
        at java.io.FileInputStream.<init>(FileInputStream.java:138)
        at java.io.FileInputStream.<init>(FileInputStream.java:93)
        at org.nativescript.staticbindinggenerator.Generator.getRows(Generator.java:123)
        at org.nativescript.staticbindinggenerator.Generator.generateBindings(Generator.java:66)
        at org.nativescript.staticbindinggenerator.Generator.writeBindings(Generator.java:50)
        at org.nativescript.staticbindinggenerator.Main.main(Main.java:15)
:asbg:generateBindings FAILED

FAILURE: Build failed with an exception.

* Where:
Build file '/nativescript-vue/samples/platforms/android/build-tools/android-static-binding-generator/build.gradle' line: 251

* What went wrong:
Execution failed for task ':asbg:generateBindings'.
> Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

this is the result of tns doctor:

```
Support for Node.js 8.1.3 is not verified. This CLI might not install or run properly.

Verifying CocoaPods. This may take more than a minute, please be patient.
  ◡ Installing iOS runtime.npm
 notice created a lockfile as package-lock.json. You should commit this file.

npm
WARN nativescript-check-cocoapods@0.0.1 No description
npm WARN nativescript-check-cocoapods@0.0.1 No repository field.
npm WARN nativescript-check-cocoapods@0.0.1 No license field.



+ tns-ios@3.1.0
added 1 package in 1.789s
  ◜ Verifying CocoaPods. This may take some time, please be patient..
Your components are up-to-date.

No issues were detected.
```

**Comment by vesper8** [comment]
After deleting the /samples/node_modules,hooks,platforms folder and running
npm cache clear --force

running npm install js-tokens from within the samples folder

now I was able to succesfully run 'tns run android --syncAllFiles' from within the /samples folder

But now getting another error, this time in the android emulator so good progress!

```
An uncaught Exception occurred on "main" thread.
java.lang.RuntimeException: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: 

Error calling module function 

Error: com.tns.NativeScriptException: Failed to find module: "../dist/index", relative to: app/
    com.tns.Module.resolvePathHelper(Module.java:146)
    com.tns.Module.resolvePath(Module.java:55)
    com.tns.Runtime.runModule(Native Method)
    com.tns.Runtime.runModule(Runtime.java:504)
    com.tns.Runtime.run(Runtime.java:499)
    com.tns.NativeScriptApplication.onCreate(NativeScriptApplication.java:17)
    android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1024)
    android.app.ActivityThread.handleBindApplication(ActivityThread.java:5403)
    android.app.ActivityThread.-wrap2(ActivityThread.java)
    android.app.ActivityThread$H.handleMessage(ActivityThread.java:1545)
    android.os.Handler.dispatchMessage(Handler.java:102)
    android.os.Looper.loop(Looper.java:154)
    android.ap
```

Any ideas?

**Comment by rigor789** [comment]
Hi, the last error is due to the recent change in directory structure. I have the same issue right now, but couldn't take a closer look at it yet. Making sure the sample app works is on my todo list though! If you in the meantime figure something out, please share! :)

**Comment by vesper8** [comment]
I was finally able to get this working. I was getting warnings about "too many symbolic links encountered". So I thought.. why are we making all these symlinks to nativescript-vue when we can just install it as a normal dependency?

So I did that. I installed "nativescript-vue": "tralves/nativescript-vue" (because I also want to fool around with .vue support)

Then I ran npm unlink from the samples folder

Had to once again delete the node_modules/platforms/hooks (not sure if all of this is required but I was getting errors again)

Then in the code inside the /samples/app folder

I played with this line (which exists 5 times)

`const Vue = require('../dist/index')`

First I tried 

`const Vue = require('../node_modules/nativescript-vue/dist/index')`

that didn't work.. tried a bunch of other things and finally I tried

`const Vue = require('nativescript-vue/dist/index')`

And vlam! That worked. The reddit app is now loading :)

Hopefully this helps you modify your instructions for a smoother install in the future.

Btw I am really excited about this project.. I hope it picks up momentum and you get more attention and contributors. I would love to port my vue web-app to a native android/ios app

I am really falling in love with React Native as well because of the ease of development with tools such as Expo and crna and very good boilerplates. But I also love vue.js and am using it in all my projects so this could potentially save a lot of time and let me reuse a lot of code.

**Comment by Kartik1607** [comment]
Having similar issue. Followed all the steps : http://imgur.com/a/wCBNG

Loop issue at step 5 was solved by moving the samples folder outside the nativescript-vue folder.

Then `tns run android` worked smoothly, but instead got error on android device.

`
An uncaught Exception occurred on "main" thread.
java.lang.RuntimeException: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: 

Error calling module function 

Error: com.tns.NativeScriptException: Failed to find module: "../dist/index", relative to: app/
    com.tns.Module.resolvePathHelper(Module.java:146)
`

Tried replacing `../dist/index` with almost all combination I could think of including the one in solution above . After failing for hours, I have given up.

**Comment by Kartik1607** [comment]
ok. got it working. finally.

Just had to add `"nativescript-vue": "tralves/nativescript-vue"` under `dependencies` in `package.json`.

Also, replace all `../dist/index` with `nativescript-vue/dist/index`

**Comment by rigor789** [comment]
@vesper8 glad you got it working, the most recent version has been published to npm, so you could get it working without cloning this repo as well. As for the boilerplates, there is a webpack boilerplate, which is being worked on. Also the guys from nuxt.js will be creating a boilerplate once a stable release is out, so that should be interesting! :) 

I will hopefully have time later this week to work on the docs, and make sure it's up to date.

@Kartik1607 glad you got it working too, all the above applies to you as well!

The template that *should* get you up and running (I haven't had the time to test it myself): 
https://www.npmjs.com/package/nativescript-vue-template

And the published `nativescript-vue` package:
https://www.npmjs.com/package/nativescript-vue

**Comment by rigor789** [comment]
I thought I would update you guys, I have changed the way samples are run. Due to the issues with npm5's link, I changed the `npm run dev` command to output to the samples folder directly. This is great for two reasons: 

- No more npm issues, since we aren't relying on symlinks and all the shenanigans
- Sample apps reload automatically when anything changes in `nativescript-vue`

I have been using this new setup for like an hour, and I love it!

I have also added a new command, which lets you run the samples more easily:

```
npm run samples
```

This will prompt you to choose the sample you want to run, using your arrow keys!!!!
And will prompt which platform to run on.

The output will be colored, to make it pretty.

So TL;DR:
```
npm run dev
npm run samples
```
